GeneratorInterface/HydjetInterface/src/HydjetHadronizer.cc

```diff
@@ -107,8 +102,14 @@ HydjetHadronizer::HydjetHadronizer(const ParameterSet& pset)
   maxEventsToPrint_ = pset.getUntrackedParameter<int>("maxEventsToPrint", 0);
   LogDebug("Events2Print") << "Number of events to be printed = " << maxEventsToPrint_;
 
-  if (embedding_)
-    src_ = pset.getParameter<edm::InputTag>("backgroundLabel");
+  if (embedding_) {
+    cflag_ = 0;
+    src_ = mayConsume<HepMCProduct>(
```

Comment:
Why `mayConsume()`? Below the `HepMCProduct` is read always when `embedding_` is `true`, so this could be just `consumes()`.

Reply:
I see, thanks.